src/mnode/ticket-processor.cpp

```diff
@@ -685,39 +696,61 @@ std::string CPastelTicketProcessor::ListFilterSellTickets(const short filter) co
                     return true;
             }
             return false;
-        });
+        }, checkConfirmation);
 }
 
-// 1 - expired;    2 - sold
-std::string CPastelTicketProcessor::ListFilterBuyTickets(const short filter) const
+// 0 - all, 1 - expired;    2 - sold
+std::string CPastelTicketProcessor::ListFilterBuyTickets(const short filter, const std::string& pastelID) const
 {
+    bool checkConfirmation = true;
+    if (filter == 0) {
+        if (pastelID.empty())
+            return ListTickets<CArtBuyTicket>(); // get all
+        else
+            checkConfirmation = false; // Get all for the pastelID, do not check confirmation
+    }
     return filterTickets<CArtBuyTicket>(
         [&](const CArtBuyTicket& t, const unsigned int chainHeight) -> bool
         {
+            if (!pastelID.empty() && t.pastelID != pastelID)
+                return true; // ignore tickets that do not belong to this pastelID
+            if (filter == 0)
+                return false; // get all belong to this pastel ID
             if (CArtTradeTicket::CheckTradeTicketExistByBuyTicket(t.GetTxId())) {
                 if (filter == 2)
                     return false; //don't skip traded
             } else if (filter == 1 && t.GetBlock() + masterNodeCtrl.MaxBuyTicketAge < chainHeight)
                 return false; //don't skip non sold, and expired
             return true;
-        });
+        }, checkConfirmation);
 }
-// 1 - available;      2 - sold
-std::string CPastelTicketProcessor::ListFilterTradeTickets(const short filter) const
+
+// 0 - all, 1 - available;      2 - sold
+std::string CPastelTicketProcessor::ListFilterTradeTickets(const short filter, const std::string& pastelID) const
 {
+    bool checkConfirmation = true;
```

Comment:
probable just write:
const bool checkConfirmation{filter > 0};
and remove
"else
checkConfirmation = false;"